### PR TITLE
[gw] fix for a concurrency issue in the leases db

### LINKED ISSUE #3259: [gw] concurrency bug in lease database: "delete statement failed: database is locked"
Publishing simultaneously can lead to errrors like

```
Lease end request - error reply: {"reason":"delete statement failed: database is locked","status":"error"}
```
This has surfaced on lhcbdev after updating to  2.10.1

### PR DESCRIPTION
**Pull request by vvolkl**
Closes #3259

### CODE REVIEW

**Comment by vvolkl** [comment]
@cernvm-bot cloudtest

File: gateway/internal/gateway/backend/lease_service.go [hunk 1]
```diff
@@ -3,11 +3,14 @@ package backend
 import (
 	"context"
 	"fmt"
+  "sync"
```
Comment (jblomer) [inline gateway/internal/gateway/backend/lease_service.go]: Whitespace seems off (also further down)

**Review by jblomer — APPROVED** [review]
LGTM, just some white spaces

**Comment by vvolkl** [comment]
@cernvm-bot cloudtest

**Pull request merged**